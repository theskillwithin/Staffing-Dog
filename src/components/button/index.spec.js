import React from 'react'
import { mount } from 'enzyme'

import Button from './index'

describe('Button', () => {
  let button
  let onClick

  beforeEach(() => {
    button = mount(<Button>My Button</Button>)
    onClick = jest.fn()
  })

  it('match snapshot', () => {
    expect(button).toMatchSnapshot()
  })

  it('should fire onclick', () => {
    button.setProps({ onClick })
    button.find('button').simulate('click')

    expect(onClick).toHaveBeenCalledTimes(1)

    onClick.mockClear()
  })
})
