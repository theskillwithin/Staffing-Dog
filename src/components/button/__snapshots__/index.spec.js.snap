// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button match snapshot 1`] = `undefined`;
